js/apps/admin-ui/src/identity-providers/add/GeneralSettings.tsx

```diff
@@ -1,19 +1,31 @@
 import { RedirectUrl } from "../component/RedirectUrl";
 import { ClientIdSecret } from "../component/ClientIdSecret";
+import { TextField } from "../component/TextField";
 import { DisplayOrder } from "../component/DisplayOrder";
 
+
 type GeneralSettingsProps = {
   id: string;
+  showclientidsecret?: boolean;
   create?: boolean;
 };
 
 export const GeneralSettings = ({
   create = true,
+  showclientidsecret = true,
   id,
-}: GeneralSettingsProps) => (
-  <>
+}: GeneralSettingsProps) => {
+  return (
+  
+    <>
     <RedirectUrl id={id} />
-    <ClientIdSecret create={create} />
+    {showclientidsecret && (
+      <>
+        <ClientIdSecret create={create} />
+      </>
```

Comment:
Since there is only a single child here, there should be no need to put this in a fragment.
```suggestion
      <ClientIdSecret create={create} />
```